pkg/controller/servicemesh/webhooks/server.go

```diff
@@ -78,5 +86,46 @@ func Add(mgr manager.Manager) error {
 		Handler: validation.NewMemberValidator(),
 	})
 
+	params = &serverParams{
+		ctx: ctx,
+		mgr: mgr,
+	}
+
+	// If SME CRD already exists, go ahead and register the SME conversion webhook right away;
+	// Don't rely on it to be registered on SMCP reconcile, as the SMCP resource might be up to
+	// date therefore the reconcile code won't be called.
+	if crdExists(ctx, mgr, webhookca.ServiceMeshExtensionCRDName) {
+		return RegisterControlPlaneWebhooks()
+	}
+
+	return nil
+}
+
+func RegisterControlPlaneWebhooks() error {
+	if params == nil {
+		log.Info("webhooks.Add() was not called or have failed. Ignoring.")
+		return nil
+	}
+
+	if params.cpwhRegistered {
```

Comment:
should we rather look at the CRD to check if we already registered a webhook?
also, does this need a mutex? we shouldn't be running >1 reconciler thread but I have some memory of @luksa working on something to that end..?